app/controllers/sessions_controller.rb

```diff
@@ -90,8 +90,15 @@ def redirect_to_origin_or_default
   end
 
   def login(options = {})
+    uid = auth_hash.uid
+    if ENV['USE_LDAP_UID_AS_EXTERNAL_ID']
+      uid_field = Rails.application.config.samson.ldap.uid
+      uid = auth_hash.extra.raw_info.send(uid_field).presence || auth_hash.uid
```

Comment:
if you already have this feature turned on, should it maybe fail if it's not set since it would not be working as intended ?

Reply:
`auth_hash.extra.raw_info.send(uid_field).presence || raise`